aion/lib/aion/room_channel/room.ex

```diff
@@ -6,13 +6,19 @@ defmodule Aion.RoomChannel.Room do
   alias Aion.RoomChannel.{Room, UserRecord}
   require Logger
 
+  @type question_list :: (list Question.t)
+
+  @type new_question_set :: %{questions: question_list, question: Question.t, answers: list Answer.t}
```

Comment:
There are a couple of issues here:
1. The preferred convention is to keep specific types in their modules. Say, you have `Aion.Room`, that means if you want to have a `Room.t` type, you will define it in `room.ex`. You are defining a new structure here, which is a set/list/collection of questions. It would be good to move it out to a separate module that would export an API for this data structure.
2. Following with the above, you introduced three names in the file: `questions`, `question_list` and  `new_question_set`. It suggests the actual implementation of each of the structures, which - afaik - is an anti-pattern. Instead you could try going with `questions` (if you have better suggestions, please tell me :joy:). If you need to store the current question, you can store it as `current_question`. 

My idea for the implementation would be that you have a list of all the questions for the game and you would just drop the head of the list every time you want to change a question. Another approach would be to keep the whole list and store an index of the current question, which you would just increment every time you want to change the question. It would not be efficient, though, as access to lists by index in Elixir is O(n).

Reply:
Your first idea is exactly what I did 😄

Reply:
Correct me if I'm wrong, but at least for me github displays that you defined two new types here :smile: and indeed, Room.t is as it should be, but it was already defined before. I only wrote it only as an example :wink: 

Those two definitions do not belong in this module:
![image](https://user-images.githubusercontent.com/15965147/31241655-e1c18530-aa04-11e7-9acb-dc66e6311cb0.png)

Reply:
I decided to remove question_list but at the same time I want to keep new_question_set. It is just an alias for structure type I use in many places

Reply:
Well, if you say you're using a structure in many places it may be an indicator that it could go to a separate module. Modules do not have to introduce a lot of logic, they may only introduce a simple struct (see `user_record.ex` in our repo for example). 

I think we could actually split the Room.ex to smaller modules dealing with specific parts of the actual game room. (so far there was no need for that, because we didn't implement a lot of logic, but eventually we will probably end up creating modules exposing APIs for parts of our Room data structure). 

You could for example try putting everything connected to questions to `Questions` module. Then you would only mention it here in the Room type spec like `questions => Questions.t`
It would represent a data structure containing both the current question and the list of preloaded question. 

Eventually, you could move all the logic connected to the questions in the game room to that module. That would make all these functions go there:
![image](https://user-images.githubusercontent.com/15965147/31249200-b69e7400-aa16-11e7-9ddb-5e6cf5dd258f.png)

What do you think about it?